teps/0154-concise-remote-resolver-syntax.md

```diff
@@ -0,0 +1,290 @@
+
+If users provide an [invalid purl string](https://github.com/package-url/purl-spec/blob/master/PURL-SPECIFICATION.rst#rules-for-each-purl-component), a validation error will be thrown in the `TaskRun/PipelineRun's` `Status`.
+
+#### Git Resolver
+Based on `git` resolver's [params](https://tekton.dev/docs/pipelines/git-resolver/#parameters), we define the following purl spec:
+
+```
+git:<git-provider>/<org>/<repo>@<revision>#<pathInRepo>?token=<token>&tokenKey=<tokenKey>&serverURL=<serverURL>&scmType=<scmType>
+```
+
+Example:
+
+```
+git:github.com/tektoncd/catalog@sha256:053a6cb9f3711d4527dd0d37ac610e8727ec0288a898d5dfbd79b25bcaa29828#task/hello-world?token=git-token
+```
+
+Using the above example, a user can do the following when referencing the Task:
+```yaml
+taskRef:
+  resolver: "git:github.com/tektoncd/catalog@sha256:053a6cb9f3711d4527dd0d37ac610e8727ec0288a898d5dfbd79b25bcaa29828#task/hello-world?token=git-token"
+```
+
+#### Bundle Resolver
+Based on `bundle` resolver's [params](https://tekton.dev/docs/pipelines/bundle-resolver/#parameters), we define the following purl spec:
+
+```
+bundle:<bundle-provider>/<bundle-namespace>/<bundle-name>@<bundle-version>?kind=<kind>&name=<name>&secret=<secret>
+``` 
+
+Example
+
+```
+bundle:docker.io/ptasci67/example-oci@sha256:053a6cb9f3711d4527dd0d37ac610e8727ec0288a898d5dfbd79b25bcaa29828?kind=task&name=hello-world?secret=default
+```
+
+Using the above example, a user can do the following when referencing the Task:
+```yaml
+taskRef:
+  resolver: "bundle:docker.io/ptasci67/example-oci@sha256:053a6cb9f3711d4527dd0d37ac610e8727ec0288a898d5dfbd79b25bcaa29828?kind=task&name=hello-world?secret=default"
+```
+
+Occasionally, users may want to fetch from a private registry (e.g. at `10.96.190.208:5000`). This host contains a `:` and violates the requirement of a purl [`type`](https://github.com/package-url/purl-spec/blob/master/PURL-SPECIFICATION.rst#rules-for-each-purl-component) and would lead to an invalid `purl`. In cases where purl is violated, users cannot use the purl `resolver-reference`. They need to use the existing verbose way of providing the `resolver params`. Providing an invalid `purl` will lead to an error.  
+
+#### Http Resolver
+Based on `http` resolver's [params](https://tekton.dev/docs/pipelines/http-resolver/#parameters), we define the following purl spec:
+
+```
+http:<schemaless-url>?http-username=<http-username>&http-password-secret=<http-password-secret>&http-password-secret-key=<http-password-secret-key>
+```
+
+In this case, the http-resolver would first attempt using `https` and then `http` (without TLS).  
```

Comment:
I don't want to pick on this if it's existing behavior, but this seems problematic 😅 

If the secure way failed we shouldn't fallback to unsecure behavior unless specifically requested.

Let users specify http vs https.

Reply:
This is not the behaviour today. Users provide the protocol they want to use as part of the url. 

I removed this approach. Thanks!